parcels/particleset/particlesetsoa.py

```diff
@@ -470,17 +562,23 @@ def add(self, particles):
         if isinstance(particles, BaseParticleSet):
             particles = particles.collection
         self._collection += particles
+        # Adding particles invalidates the neighbor search structure.
+        self._dirty_neighbor = True
         return self
 
     def remove_indices(self, indices):
         """Method to remove particles from the ParticleSet, based on their `indices`"""
+        # Removing particles invalidates the neighbor search structure.
+        self._dirty_neighbor = True
         if type(indices) in [int, np.int32, np.intp]:
             self._collection.remove_single_by_index(indices)
         else:
             self._collection.remove_multi_by_indices(indices)
 
     def remove_booleanvector(self, indices):
         """Method to remove particles from the ParticleSet, based on an array of booleans"""
+        # Removing particles invalidates the neighbor search structure.
+        self._dirty_neighbor = True
         self.remove_indices(np.where(indices)[0])
 
```

Comment:
I think the modified execute-function should have a place in here. Perhaps even as in the following (pseudocode):

```
def execute(...):
    if self.interaction_kernel is None:
        super(ParticleSetSOA, self).execute(....)
    else:
        self.execute_with_interaction(...)

def execute_with_interaction(...):
    <here comes the execute-function code that currently is in the BaseParticleSet>
```